Kernel/Storage/Partition/DiskPartitionMetadata.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <AK/RefPtr.h>
+#include <Kernel/Devices/BlockDevice.h>
+
+namespace Kernel {
+
+class DiskPartitionMetadata {
+public:
+    DiskPartitionMetadata(u64 block_offset, u64 block_limit, ByteBuffer partition_type);
+    DiskPartitionMetadata(u64 block_offset, u64 block_limit, ByteBuffer partition_type, ByteBuffer unique_guid, u64 special_attributes, String name);
+    u64 start_block() const;
+    u64 end_block() const;
+
+    Optional<u64> special_attributes() const;
+    Optional<String> name() const;
+    Optional<ByteBuffer> partition_type() const;
+    Optional<ByteBuffer> unique_guid() const;
+
+private:
+    u64 m_start_block;
+    u64 m_end_block;
+    ByteBuffer m_partition_type;
+    ByteBuffer m_unique_guid;
+    u64 m_attributes;
```

Comment:
Please always initialize primitive members explicitly.
```cpp
u64 m_attributes { 0 };
```
Notice how one of the constructors forgets to initialize this member :)

Reply:
Thanks for catching this, I'll try to pay attention to such things later on :)